src/marqo/tensor_search/tensor_search.py

```diff
@@ -267,9 +267,53 @@ def _autofill_index_settings(index_settings: dict):
 
 
 def get_stats(config: Config, index_name: str):
-    doc_count = HttpRequests(config).post(path=F"{index_name}/_count")["count"]
+    """Returns the number of documents and vectors in the index.
+
+    The _count API counts top-level documents.
+    For numberOfVectors, we count the number of `__chunks.__field_name` fields because it is a key_word field, which
+    is known to be fast.
+
+    Difference between the two gives the numberOfVectors."""
+
+    body = {
+        "size": 0,
+        "aggs": {
+            "nested_chunks": {
+                "nested": {
+                    "path": "__chunks"
+                },
+                "aggs": {
+                    "marqo_vector_count": {
+                        "value_count": {
+                            "field": "__chunks.__field_name"
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    try:
+        doc_count = HttpRequests(config).post(path=F"{index_name}/_count")["count"]
+        vector_count = HttpRequests(config).get(path=f"{index_name}/_search", body=body) \
+            ["aggregations"]["nested_chunks"]["marqo_vector_count"]["value"]
+    except (KeyError, TypeError) as e:
+        error_message = (f"Marqo received an unexpected response from Marqo-os during execution of `get_stats()` APIs. "
+                         f"The expected fields do not exist in the response. Original error message = {e}")
+        logger.error(error_message)
+        raise errors.InternalError(error_message)
+    except (errors.IndexNotFoundError, errors.InvalidIndexNameError):
+        raise
```

Comment:
This `except` seems superfluous. You're not doing anything with these error types.

Reply:
I do this because IndexNotFoundError and InvalidIndexNameError are inherited from MarqoWebError.  For these two errors, I want to raise it. For other errors, i want to convert them to internal errors.

Reply:
I updated the code to make it more straightforward